tests/test_routes.py

```diff
@@ -91,3 +94,117 @@ def test_index(self):
+        )
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        data = resp.get_json()
+        self.assertEqual(data["id"], test_shopcart.id)
+
+    def test_get_shopcart_not_found(self):
+        """It should not Read a Shopcart that is not found"""
+        response = self.client.get(f"{BASE_URL}/0")
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+    
+    # @patch.object(Shopcart, 'get_by_id', MagicMock(side_effect=Exception("DBAPIErr")))
+    # def test_list_shopcart_items_shopcart_get_by_id_error(self):
+    #     """ It should get internal server error if there's exception in Shopcart.get_by_id """
+    #     resp = self.client.get(f"{BASE_URL}/0/items")
+    #     self.assertEqual(resp.status_code, status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+
+    # def test_create_shopcart(self):
+    #     """It should Create a new Shopcart"""
+    #     shopcart = ShopcartFactory()
+    #     resp = self.client.post(
+    #         BASE_URL, json=shopcart.serialize(), content_type="application/json"
+    #     )
+    #     self.assertEqual(resp.status_code, status.HTTP_201_CREATED)
+
+    #     # Make sure location header is set
+    #     location = resp.headers.get("Location", None)
+    #     self.assertIsNotNone(location)
+
+    #     # Check the data is correct
+    #     new_shopcart = resp.get_json()
+    #     self.assertEqual(new_shopcart["name"], shopcart.name, "Names does not match")
+        
+
+    #     # Check that the location header was correct by getting it
+    #     resp = self.client.get(location, content_type="application/json")
+    #     self.assertEqual(resp.status_code, status.HTTP_200_OK)
+    #     new_shopcart = resp.get_json()
+    #     self.assertEqual(new_shopcart["name"], shopcart.name, "Names does not match")
+    
+
+    # def test_create_shopcarts(self):
+    #     """ It should create an shopcart """
+    #     shopcart = ShopcartFactory()
+    #     resp = self.client.post(
+    #         f"{BASE_URL}",json = shopcart.serialize(), content_type = "application/json"
+    #     )
+        
+    #     self.assertEqual(resp.status_code, status.HTTP_201_CREATED)
+        
+
+    #     location = resp.headers.get("location", None)
+    #     self.assertIsNotNone(location)
+
+    #     new_shopcart = resp.get_json()
+    #     # self.assertEqual(data["date"], shopcart.date,"date does not match")
+    #     self.assertEqual(new_shopcart["name"],shopcart.name, "name does not match")
+        
+
+    def test_create_shopcart_missing_info(self):
+        """
+        It should fail if the call has some missing information.
+        """
+        resp = self.client.post(
+            f"{BASE_URL}",
+            json={
+                "items": []
+            },
+            content_type="application/json",
+        )
+        self.assertEqual(resp.status_code, status.HTTP_400_BAD_REQUEST)
+
+       
+
+    # def test_get_item(self):
```

Comment:
could you help remove the commented code lines not in used below?

Reply:
done